File: Client/VRIS/app/src/main/java/com/valtech/amsterdam/vris/model/Room.java
package com.valtech.amsterdam.vris.model;

/**
 * Created by jasper.van.zijp on 14-7-2017.
 */

import com.google.gson.annotations.SerializedName;
import com.valtech.amsterdam.recyclist.annotation.ApiInfo;
import com.valtech.amsterdam.recyclist.modifiers.IHasId;

/**
 * A room registered in the Office namespace
 * Todo: unused
 */
@ApiInfo(methodName = "room")
public class Room implements IHasId {
    @SerializedName("Id") private int mId;
    @SerializedName("Name")private String mName;

    /**
     * Initiate a new room
     * @param id
     * @param name
     */
    public Room(int id, String name) {
        mId = id;
        mName = name;
    }

    /**
     * Get the id of the room
     * @return
     */
    public int getId() {
        return mId;
    }

    /**
     * Get the rooms name
     * @return
     */
    public String getName() { return mName; }
}

File: Client/app/src/main/java/com/labs/valtech/vris/models/Room.kt
package com.labs.valtech.vris.models

import com.google.gson.annotations.SerializedName
import com.labs.valtech.vris.R
import com.labs.valtech.vris.VrisApplication


/**
 * Created by jasper.van.zijp on 18-7-2017.
 */

data class Room (
        @field:SerializedName("id") override var id: String = "",
        @field:SerializedName("venue") override var venue: String = "",
        @field:SerializedName("country") override var country: String = "",
        @field:SerializedName("name") override var name: String = "") : IRoom {

    override fun getFullRoomName(): String {
        return "$country $venue $name"
    }
}